src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1848,3 +1848,9 @@ def validate_policy(namespace):
     if namespace.id is not None:
         logger.warning("\nPlease do not specify --expiry and --permissions if they are already specified in your "
                        "policy.")
+
+
+def validate_blob_name_for_upload(namespace):
+    if not namespace.blob_name:
+        namespace.blob_name = namespace.file_path.split('/')[-1].split('\\')[-1]
```

Comment:
[`os.path.basename`](https://docs.python.org/3/library/os.path.html#os.path.basename) is a simpler solution.

Reply:
Got it. Done.